cmd/main.go

```diff
@@ -291,7 +296,8 @@ func setupKymaReconciler(mgr ctrl.Manager, descriptorProvider *provider.CachedDe
 	moduleTemplateInfoLookupStrategies := moduletemplateinfolookup.NewModuleTemplateInfoLookupStrategies([]moduletemplateinfolookup.ModuleTemplateInfoLookupStrategy{
 		moduletemplateinfolookup.NewByVersionStrategy(mgr.GetClient()),
 		moduletemplateinfolookup.NewByChannelStrategy(mgr.GetClient()),
-		moduletemplateinfolookup.NewByModuleReleaseMetaStrategy(mgr.GetClient()),
+		moduletemplateinfolookup.NewWithMaintenanceWindowDecorator(maintenanceWindow,
+			moduletemplateinfolookup.NewByModuleReleaseMetaStrategy(mgr.GetClient())),
```

Comment:
I have only enabled the maintenance window decorator for the new module metadata. We can discuss in the daily if we want to enable it for old module metadata as well. If so, we need to invest some more time to also e2e test these scenarios. I have moved the decision to the rollout issue: https://github.com/kyma-project/lifecycle-manager/issues/2165